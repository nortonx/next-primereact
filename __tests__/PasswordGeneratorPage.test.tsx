import { render, screen, fireEvent } from "@testing-library/react"
import PasswordGeneratorPage from "@/app/password-generator/page"
import { describe, it, expect } from "@jest/globals"

describe("Password Generator Page", () => {
  it("renders page and match snapshot", async () => {
    const { container } = render(<PasswordGeneratorPage/>)
    const page = await screen.findByTestId("password-generator-page")
    expect(page).toBeDefined()
    expect(container).toMatchSnapshot()
  })

  it("should generate password when 'Generate Password' button is clicked", async () => {
    render(<PasswordGeneratorPage/>) 
    const passwordField = await screen.findByTestId("password-field")
    expect(passwordField.nodeValue).toBeFalsy()
    const generatePasswordBtn = await screen.findByTestId("generate-password-btn")
    fireEvent.click(generatePasswordBtn)
    expect(passwordField.nodeValue).toBeDefined()
  })

  it("should clear password field when the button 'Clear Password' is clicked", async () => { 
    render(<PasswordGeneratorPage/>) 
    const passwordField = await screen.findByTestId("password-field")
    const generatePasswordBtn = await screen.findByTestId("generate-password-btn")
    fireEvent.click(generatePasswordBtn)
    expect(passwordField.nodeValue).toBeDefined()
    const clearPasswordBtn = await screen.findByTestId("clear-password-btn")
    fireEvent.click(clearPasswordBtn)
    expect(passwordField.nodeValue).toBeFalsy()
  })

  it("should copy password to clipboard when the button 'Copy Password' is clicked", async () => {
    render(<PasswordGeneratorPage/>) 
    const passwordField = await screen.findByTestId("password-field")
    const generatePasswordBtn = await screen.findByTestId("generate-password-btn")
    fireEvent.click(generatePasswordBtn)
    expect(passwordField.nodeValue).toBeDefined()
    const copyPasswordBtn = await screen.findByTestId("copy-password-btn")
    fireEvent.click(copyPasswordBtn)
    const toast = await screen.findByTestId("toast")
    expect(toast).toBeDefined()
    expect(passwordField.nodeValue).toBeDefined()
  })

  // it("should display a toast message when the button 'Copy Password' is clicked", async () => {
  //   render(<PasswordGeneratorPage/>)
  //   const generatePasswordBtn = await screen.findByTestId("generate-password-btn")
  //   fireEvent.click(generatePasswordBtn)
  //   const copyPasswordBtn = await screen.findByTestId("copy-password-btn")
  //   fireEvent.click(copyPasswordBtn)
  //   const toast = await screen.findByTestId("toast")
  //   expect(toast).toBeDefined()
  // })
})