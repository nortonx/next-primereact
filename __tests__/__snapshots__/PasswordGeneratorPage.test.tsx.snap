// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Password Generator Page renders page and match snapshot 1`] = `
<div>
  <main
    class="surface-0"
    data-testid="password-generator-page"
  >
    <div
      class="text-900 font-bold text-6xl mb-4 text-center"
    >
      Password Generator
    </div>
    <div
      class="grid"
    >
      <div
        class="col-12 lg:col-4"
      />
      <div
        class="col-12 lg:col-4"
      >
        <div
          class="p-3 h-full"
        >
          <div
            class="shadow-2 p-3 h-full flex flex-column"
          >
            <input
              class="password p-inputtext p-component"
              data-pc-name="inputtext"
              data-pc-section="root"
              data-testid="password-field"
              value=""
            />
            <button
              aria-label="Copy password"
              class="mt-3 p-button p-component"
              data-pc-name="button"
              data-pc-section="root"
            >
              <span
                class="p-button-icon p-c p-button-icon-left pi pi-copy"
                data-pc-section="icon"
              />
              <span
                class="p-button-label p-c"
                data-pc-section="label"
              >
                Copy password
              </span>
            </button>
            <div
              class="flex align-items-center mt-3"
            >
              <div
                class="p-checkbox p-component"
                data-p-disabled="false"
                data-p-highlight="false"
                data-pc-name="checkbox"
                data-pc-section="root"
              >
                <input
                  aria-invalid="false"
                  class="p-checkbox-input"
                  data-pc-section="input"
                  id="uppercaseLetters"
                  type="checkbox"
                />
                <div
                  class="p-checkbox-box"
                  data-p-disabled="false"
                  data-p-highlight="false"
                  data-pc-section="box"
                />
              </div>
              <label
                class="ml-2"
                for="uppercaseLetters"
              >
                Enable Uppercase Letters
              </label>
            </div>
            <div
              class="flex align-items-center mt-3"
            >
              <div
                class="p-checkbox p-component"
                data-p-disabled="false"
                data-p-highlight="false"
                data-pc-name="checkbox"
                data-pc-section="root"
              >
                <input
                  aria-invalid="false"
                  class="p-checkbox-input"
                  data-pc-section="input"
                  id="numbers"
                  type="checkbox"
                />
                <div
                  class="p-checkbox-box"
                  data-p-disabled="false"
                  data-p-highlight="false"
                  data-pc-section="box"
                />
              </div>
              <label
                class="ml-2"
                for="numbers"
              >
                Enable Numbers
              </label>
            </div>
            <div
              class="flex align-items-center mt-3"
            >
              <div
                class="p-checkbox p-component"
                data-p-disabled="false"
                data-p-highlight="false"
                data-pc-name="checkbox"
                data-pc-section="root"
              >
                <input
                  aria-invalid="false"
                  class="p-checkbox-input"
                  data-pc-section="input"
                  id="specialCharacters"
                  type="checkbox"
                />
                <div
                  class="p-checkbox-box"
                  data-p-disabled="false"
                  data-p-highlight="false"
                  data-pc-section="box"
                />
              </div>
              <label
                class="ml-2"
                for="specialCharacters"
              >
                Enable Special Characters
              </label>
            </div>
            <div
              class="flex align-items-center mt-3 ml-2 justify-content-center"
            >
              <div
                class="p-knob p-component"
                data-pc-name="knob"
                data-pc-section="root"
              >
                <svg
                  aria-valuemax="36"
                  aria-valuemin="8"
                  aria-valuenow="8"
                  data-pc-section="svg"
                  height="100"
                  role="slider"
                  tabindex="0"
                  viewBox="0 0 100 100"
                  width="100"
                >
                  <path
                    class="p-knob-range"
                    d="M 29.999999999999982 84.64101615137754 A 40 40 0 1 1 70 84.64101615137754"
                    data-pc-section="range"
                    stroke="var(--surface-border, LightGray)"
                    stroke-width="14"
                  />
                  <path
                    class="p-knob-value"
                    d="M 29.999999999999982 84.64101615137754 A 40 40 0 0 1 29.999999999999982 84.64101615137754"
                    data-pc-section="value"
                    stroke="var(--primary-color, Black)"
                    stroke-width="14"
                  />
                  <text
                    class="p-knob-text"
                    data-pc-section="label"
                    fill="var(--text-color-secondary, Black)"
                    text-anchor="middle"
                    x="50"
                    y="57"
                  >
                    8
                  </text>
                </svg>
              </div>
            </div>
            <div
              class="flex align-item-center mt-3 justify-content-around"
            >
              <button
                aria-label="Generate Password"
                class="mr-1 p-button p-component"
                data-pc-name="button"
                data-pc-section="root"
                data-testid="generate-password-btn"
              >
                <span
                  class="p-button-icon p-c p-button-icon-left pi pi-replay"
                  data-pc-section="icon"
                />
                <span
                  class="p-button-label p-c"
                  data-pc-section="label"
                >
                  Generate Password
                </span>
              </button>
              <button
                aria-label="Clear Password"
                class="p-button p-component"
                data-pc-name="button"
                data-pc-section="root"
              >
                <span
                  class="p-button-icon p-c p-button-icon-left pi pi-ban"
                  data-pc-section="icon"
                />
                <span
                  class="p-button-label p-c"
                  data-pc-section="label"
                >
                  Clear Password
                </span>
              </button>
              <div
                class="p-toast p-component p-toast-top-right p-ripple-disabled"
                data-pc-name="toast"
                data-pc-section="root"
                style="position: fixed; top: 20px; right: 20px;"
              >
                <div />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </main>
</div>
`;
